### PR TITLE
Change OSTYPE for target_platform

### PR DESCRIPTION
**Pull request by AntoinePrv**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] ~Reset the build number to `0` (if the version changed)~
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by AntoinePrv** [comment]
@conda-forge-admin, please rerender

File: .scripts/run_docker_build.sh [hunk 1]
```diff
@@ -65,7 +69,9 @@ DOCKER_RUN_ARGS="${CONDA_FORGE_DOCKER_RUN_ARGS}"
 if [ -z "${CI}" ]; then
     DOCKER_RUN_ARGS="-it ${DOCKER_RUN_ARGS}"
 fi
+endgroup "Configure Docker"
 
+startgroup "Start Docker"
```
Comment (fschloesser) [inline .scripts/run_docker_build.sh:74]: did you miss to put the `endgroup` line or is that intended?

File: .scripts/build_steps.sh [hunk 2]
```diff
@@ -6,8 +6,14 @@
 # benefit from the improvement.
 
 set -xeuo pipefail
-export PYTHONUNBUFFERED=1
 export FEEDSTOCK_ROOT="${FEEDSTOCK_ROOT:-/home/conda/feedstock_root}"
+source ${FEEDSTOCK_ROOT}/.scripts/logging_utils.sh
+
+
+endgroup "Start Docker"
```
Comment (fschloesser) [inline .scripts/build_steps.sh:13]: why is the first think that is logged a `endgroup`, should this by any chance be a `startgroup`? (I don't know a lot about the output that is needed in the logging of the different tools)

File: .scripts/build_steps.sh [hunk 3]
```diff
@@ -18,8 +24,9 @@ conda-build:
  root-dir: ${FEEDSTOCK_ROOT}/build_artifacts
 
 CONDARC
+BUILD_CMD=build
 
-conda install --yes --quiet "conda-forge-ci-setup=3" conda-build pip -c conda-forge
+conda install --yes --quiet "conda-forge-ci-setup=3" conda-build pip ${GET_BOA:-} -c conda-forge
```
Comment (fschloesser) [inline .scripts/build_steps.sh:29]: Where does the `GET_BOA` variable get defined and what does it do?

File: .scripts/build_steps.sh [hunk 4]
```diff
@@ -6,8 +6,14 @@
 # benefit from the improvement.
 
 set -xeuo pipefail
-export PYTHONUNBUFFERED=1
 export FEEDSTOCK_ROOT="${FEEDSTOCK_ROOT:-/home/conda/feedstock_root}"
+source ${FEEDSTOCK_ROOT}/.scripts/logging_utils.sh
```
Comment (fschloesser) [inline .scripts/build_steps.sh:10]: why do you need `FEEDSTOCK_ROOT` here but not in the other scripts where you source the logging?

**Review by fschloesser — COMMENTED** [review]
You added tbb and improved logging :+1: 
Other than changing OSTYPE for target_platform like mentioned in the title you have done some small additional changes. I think I'm missing a bit of context here, so I commented on some that I need some filling in. 
Overall the pull request looks good and the builds seem to succeed :)

**Comment by AntoinePrv** [comment]
Thank you @fschloesser and sorry for not providing more context.

Basically, I changed `OSTYPE` for `target_platform` as it was suggested to me in another PR, because technically, conda could be cross compiling. I don't believe it makes a difference here.

For the rest, everything that is not under `recipe` (all the files you commented on) are automatically generated by conda-forge through the `@conda-forge-admin, please rerender`.
They have to do with deployment scripts, and version pinning of our dependencies (so that we play nice with other packages).
I also don't understand all their details, but we don't have to.

**Pull request merged**